crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -478,6 +498,20 @@ pub struct ReadOnlyOwnUserIdentity {
     verified: Arc<AtomicBool>,
 }
 
+/// Specific equality check for identities that also check if there are changes
+/// in signatures on the master cross signing key. The default implementation
+/// for MasterPubkey ignores signatures.
+impl PartialEq for ReadOnlyOwnUserIdentity {
+    fn eq(&self, other: &Self) -> bool {
+        self.user_id == other.user_id
+            && self.master_key == other.master_key
+            && self.self_signing_key == other.self_signing_key
+            && self.user_signing_key == other.user_signing_key
+            && self.is_verified() == other.is_verified()
+            && self.master_key.signatures() == other.master_key.signatures()
+    }
+}
```

Comment:
I kinda wonder if we should define a specific (private) method for this, rather than using ==, not least because we know that `is_verified` and `user_id` are not changing.

Not sure though.

Reply:
When discussing with poljar it was decided to go for PartialEq instead of custom method.

Reply:
The type is supposed to be private anyways (I know it realistically isn't but that's mostly because the stores need to have access to them), so we can as well use `PartialEq` and document the what the implementation does.